cpp/src/arrow/util/memory-pool-test.cc

```diff
@@ -46,4 +46,14 @@ TEST(DefaultMemoryPool, OOM) {
   ASSERT_RAISES(OutOfMemory, pool->Allocate(to_alloc, &data));
 }
 
+TEST(DefaultMemoryPoolDeathTest, FreeLargeMemory) {
+  MemoryPool* pool = default_memory_pool();
+
+  uint8_t* data;
+  ASSERT_OK(pool->Allocate(100, &data));
+  EXPECT_EXIT(pool->Free(data, 120), ::testing::ExitedWithCode(1),
+               ".*Check failed: \\(bytes_allocated_\\) >= \\(size\\)");
+  pool->Free(data, 100);
```

Comment:
Thank you for your comment. Actually, I'm not an expert of C++, and would like to ask a question which may sound silly. 
Does your comment mean that this unit test needs to be disabled for release builds? or, DCHECK in Free() should be disabled? 
Would you please give me some backgrounds around this briefly?